deps/kspec.jl

```diff
@@ -57,18 +61,20 @@ optimization flag and `FOO=yes` is included in the environment variables.
 
 The returned `kernelpath` is the path of the installed kernel directory, something like `/...somepath.../kernels/julia-O3-1.0`
 (in Julia 1.0).  The `specname` argument can be passed to alter the name of this
-directory (which defaults to `name` with spaces replaced by hyphens).
+directory (which defaults to `name` with spaces replaced by hyphens), and the `specversion`
+argument to alter the _version_ part of this name, which gets appended to the `specname`.
 
 You can uninstall the kernel by calling `rm(kernelpath, recursive=true)`.
 """
 function installkernel(name::AbstractString, julia_options::AbstractString...;
                    specname::AbstractString = replace(lowercase(name), " "=>"-"),
+                   # TODO: Should debugdesc() be part of the user-overridable specversion,
+                   # or should it always be appended as it was before?
+                   specversion::AbstractString = "$(VERSION.major).$(VERSION.minor)$(debugdesc())",
```

Comment:
This TODO is my main remaining question.

I think we wouldn't want it to be always appended like it was before, for the same reasons as the original motivation for this PR (https://github.com/JuliaLang/IJulia.jl/issues/852) -- if we tried to run a CI test w/ a debug julia build, the notebook tests would fail.

So perhaps the `debugdesc` should also become its own separate parameter as well? I think leaving it as I've written it seems good enough, but happy to change it if you think that's best.

Reply:
I've separated this to be its own parameter as well.